resources/src/trade.rs

```diff
@@ -299,3 +299,100 @@ mod trade_tests {
         );
     }
 }
+
+/// The aggregation of trades for a specifc base/counter pair of assets over a given
+/// time period.
+#[derive(Deserialize, Serialize, Debug)]
+pub struct TradeAggregation {
+    // Several fields are omitted since they don't seem to be in the actual response from horizon.
+    // Or they don't seem to make sense.
+    //
+    // The rational versions of the prices also only seem to be present in the examples. When
+    // querying the actual horizon API they were absent. They probably don't contain much value
+    // over the "amount" field anyhow so I left them off.
+    timestamp: u64,
+    trade_count: u64,
+    base_volume: Amount,
+    counter_volume: Amount,
+    avg: Amount,
+    high: Amount,
+    low: Amount,
+    open: Amount,
+    close: Amount,
+}
+
+impl TradeAggregation {
+    /// The beginning of this time segment
+    pub fn started_at(&self) -> DateTime<Utc> {
```

Comment:
Good call :)

Reply:
yea, i could translate inside the serde itself, which may be a bit faster since it only needs to happen once, but this seemed fine for now since it's just integer math.